### PR TITLE
blobtx: add new flags to override blobs' params & bug fix;

### PR DESCRIPTION
**Pull request by GalaIO**
### Description

blobtx: add new flags to override blobs' params & bug fix;

### Changes

Notable changes: 
* flags: add new flags to override blobs' params;
* freezer: fix blob ancient save error;

### CODE REVIEW

File: core/rawdb/chain_freezer.go [hunk 1]
```diff
@@ -292,14 +292,66 @@ func getBlobExtraReserveFromEnv(env *ethdb.FreezerEnv) uint64 {
...
+	preHashes, err := f.freezeRange(nfdb, number, cancunNumber-1, env)
+	if err != nil {
+		return preHashes, err
+	}
+
+	if err = ResetEmptyBlobAncientTable(f, cancunNumber); err != nil {
+		return preHashes, err
+	}
+	// freeze post cancun
+	postHashes, err := f.freezeRange(nfdb, cancunNumber, limit, env)
```
Comment (zzzckck) [inline core/rawdb/chain_freezer.go:345]: Q1: what's the difference between "freeze post cancun" and "freeze pre cancun"?

Comment (zzzckck) [reply core/rawdb/chain_freezer.go:345]: A: mainly to do "ResetEmptyBlobAncientTable" before "freeze post cancun".

**Review by zzzckck — APPROVED** [review]

**Review by coinshowmaker — APPROVED** [review]

**Review by buddh0 — APPROVED** [review]

File: cmd/utils/flags.go [hunk 2]
```diff
@@ -315,8 +315,25 @@ var (
...
+	OverrideFullImmutabilityThreshold = &cli.Uint64Flag{
+		Name:     "override.immutabilitythreshold",
+		Usage:    "It is the number of blocks after which a chain segment is considered immutable, only testing purple before Cancun",
+		Value:    params.FullImmutabilityThreshold,
+		Category: flags.EthCategory,
+	}
+	OverrideMinBlocksForBlobRequests = &cli.Uint64Flag{
+		Name:     "override.minforblobrequest",
+		Usage:    "It keeps blob data available for min blocks in local, only testing purple before Cancun",
+		Value:    params.MinBlocksForBlobRequests,
```
Comment (buddh0) [inline cmd/utils/flags.go:330]: purpose?

File: cmd/utils/flags.go [hunk 3]
```diff
@@ -315,8 +315,25 @@ var (
...
 	}
 	OverrideFeynmanFix = &cli.Uint64Flag{
-		Name:     "override.feynmanfix",
-		Usage:    "Manually specify the FeynmanFix fork timestamp, overriding the bundled setting",
+		Name:  "override.feynmanfix",
+		Usage: "Manually specify the FeynmanFix fork timestamp, overriding the bundled setting",
+	}
+	OverrideFullImmutabilityThreshold = &cli.Uint64Flag{
+		Name:     "override.immutabilitythreshold",
+		Usage:    "It is the number of blocks after which a chain segment is considered immutable, only testing purple before Cancun",
```
Comment (buddh0) [inline cmd/utils/flags.go:323]: I think we can keep them even after Cancun
just note that it's  only for testing purpose

**Pull request merged**